tests/system_tests_expandvars.py

```diff
@@ -0,0 +1,91 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import os
+import sys
+import subprocess
+from subprocess import STDOUT, CalledProcessError
+import shutil
+from system_test import TestCase
+
+env_config = {
+    'PORT': '12345',
+    'NAME': 'YoYoMa'
+}
+
+current_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)))
+path_parent = os.path.dirname(os.path.join(os.path.dirname(os.path.abspath(__file__))))
+expandvars_script_folder = os.path.join(path_parent, 'scripts')
+script_file = "expandvars.py"
+env = dict(os.environ)
+env.update(env_config)
+
+
+class ExpandvarsTest(TestCase):
+    def test_expandvars_input_file_output_file(self):
+        # Provide an input file and an output file to the expandvars.py script
+        # and make sure the output file is populated with the expanded vars.
+        nginx_conf_dir = os.path.join(current_dir, 'nginx-configs')
+        input_file = os.path.join(nginx_conf_dir, 'expandvars.conf')
+        output_file = os.path.join(self.directory, 'expandvars.conf')
+        args = [sys.executable, os.path.join(expandvars_script_folder, script_file), input_file, output_file]
+        subprocess.run(args, stderr=STDOUT, check=True, env=env)
+        port_expanded = False
+        name_expanded = False
+        with open(output_file) as ofile:
+            for line in enumerate(ofile):
+                test_line = line[1].strip()
+                if "PORT=12345" == test_line:
+                    port_expanded = True
+                if "NAME=YoYoMa" == test_line:
+                    name_expanded = True
+        self.assertTrue(port_expanded and name_expanded)
```

Comment:
This is way too circumspect approach, I feel.

First, `for line in enumerate(ofile):` should better be `for line_number, line in enumerate(ofile, start=1):` due to how https://docs.python.org/3/library/functions.html#enumerate works.

What about something like this (I did not run it, just an idea)?

```
with open(output_file) as ofile:
    lines = ofile.readlines()
self.assertIn("PORT=12345\n",  lines)
self.assertIn("NAME=YoYoMa\n",  lines)
```

maybe get rid of the \n's (if they are present, I think they are), for that, `lines = ofile.read().splitlines()` should work.

Reply:
I removed the enumeration but I am still looping for simplicity

Reply:
The simplest and most direct assertion would be to assert on the file content as a string, I'd say, no looping necessary.

```
with open(output_file) as ofile:
    self.assertEquals("PORT=12345\nNAME=YoYoMa\n", ofile.read())
```

This might have a problem with platform dependent line-ending characters, but I _believe_ it won't be problem here.

Reply:
I would like to avoid using "\n" in the code. I think what I have now is pretty simple to follow. Thx

Reply:
It is not too bad, but I still like my version better ;)

PR approved!